compiler/next/include/chpl/uast/Use.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright 2021 Hewlett Packard Enterprise Development LP
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CHPL_UAST_USE_H
+#define CHPL_UAST_USE_H
+
+#include "chpl/queries/Location.h"
+#include "chpl/uast/Decl.h"
+#include "chpl/uast/Expression.h"
+#include "chpl/uast/UseClause.h"
+
+namespace chpl {
+namespace uast {
+
+
+/**
+  This class represents a use statement. For example:
+
+  \rst
+  .. code-block:: chapel
+
+      // Example 1:
+      use Foo, Bar as A;
+
+  \endrst
+
+  This creates a use statement that has two use clauses, 'Foo' and
+  'Bar as A'.
+*/
+class Use final : public Expression {
```

Comment:
There is probably some reason you didn't do this but I thought we were considering having `Use` inherit from `Decl` ? Is your point that it's not really declaring anything (rather just making something available or possibly renaming something?). I think we could view them as declaring something but it's more about the exact meaning of "declaration" that we use than anything deeper.

Anyway I am not worried about it because the follow-on resolution steps will need to treat Use/Import directly since they affect what symbols are available so much.

Reply:
I ended up not doing that because it left more questions than answers.